### PR TITLE
feat: add web page to query artifacts

### LINKED ISSUE #99: Create a simple HTML page for snark-artifacts.pse.dev
**Description**

The base URL [snark-artifacts.pse.dev](https://snark-artifacts.pse.dev) is used for downloading artifacts. The main page may however include an `index.html` file to present the project and give some basic information.

### PR DESCRIPTION
**Pull request by sripwoud**
Add web page (we can improve on the css styling later ^^) to use as s3 bucket main index.html to display at `https://snark-artifacts.pse.dev`

image

Close #99

## Test:
`pn --filter web dev`, interact with the page to download artifacts

### CODE REVIEW

File: .github/workflows/deploy-web-app.yaml
Comment (sripwoud) [inline .github/workflows/deploy-web-app.yaml:1]: @ntampakas could you check this?
I am not sure what is the canonical way to deploy a react app to a s3 bucket...

to build the app: install deps etc... then `pnpm build`, we get an `index.html` and some `assets/` (js, css) in a `apps/web/dist`.
Is it enough to copy this over the s3 bucket root path ?

File: apps/package.json
Comment (cedoor) [inline apps/package.json:1]: Is this file necessary?

**Review by cedoor — APPROVED** [review]
Great :)

Should we add an option to download all the artifacts of a specific version?

File: apps/package.json
Comment (sripwoud) [reply apps/package.json:1]: no

File: .github/workflows/deploy-web-app.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: arn:aws:iam::490752553772:role/snark-artifacts-assume_role-slc
+          role-duration-seconds: 900
+          aws-region: us-west-2
+
+      - name: Upload index.html and assets to s3 bucket
+        run: |
+          aws s3 sync apps/web/dist s3://snark-artifacts/ --delete
```
Comment (ntampakas) [inline .github/workflows/deploy-web-app.yaml:56]: This command affects all files in the S3 bucket. I suggest changing it to this:
```suggestion
          aws s3 sync apps/web/dist s3://snark-artifacts/ --exclude "*" --include "index.html" --include "assets/*" --delete
```

File: .github/workflows/deploy-web-app.yaml
Comment (ntampakas) [reply .github/workflows/deploy-web-app.yaml:1]: Sure, i' ll check it ASAP!
The logic behind this implementation is 100% correct!

**Review by ntampakas — APPROVED** [review]
LGTM!
